File: src/app/Models/RawMaterialEntry.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use App\Traits\Auditable;

class RawMaterialEntry extends Model
{
    use HasFactory, SoftDeletes, Auditable;

    protected $fillable = [
        'raw_material_type_id',
        'provider_id',
        'raw_material_characteristic_id',
        'entry_number',
        'remito',
        'batch',
        'entry_date',
        'quantity_kg',
        'coils_count',
        'status',
        'observations'
    ];

    protected $casts = [
        'entry_date' => 'date',
        'quantity_kg' => 'decimal:3',
    ];

    public function type()
    {
        return $this->belongsTo(RawMaterialType::class, 'raw_material_type_id');
    }

    public function provider()
    {
        return $this->belongsTo(Provider::class);
    }

    public function characteristic()
    {
        return $this->belongsTo(RawMaterialCharacteristic::class, 'raw_material_characteristic_id');
    }

    public function test()
    {
        return $this->hasOne(MaterialTest::class);
    }
}

File: backend-dotnet/src/Ikhana.Domain/Entities/RawMaterialEntry.cs
using System.ComponentModel.DataAnnotations.Schema;
using Ikhana.Domain.Enums;

namespace Ikhana.Domain.Entities;

public class RawMaterialEntry : BaseEntity, ISoftDelete, IAuditable
{
    public long RawMaterialTypeId { get; set; }
    public long ProviderId { get; set; }
    public long RawMaterialCharacteristicId { get; set; }

    public int? EntryNumber { get; set; }
    public string? Remito { get; set; }
    public int? Batch { get; set; }
    public DateOnly EntryDate { get; set; }
    public decimal? QuantityKg { get; set; }
    public int? CoilsCount { get; set; }

    public EntryStatus Status { get; set; } = EntryStatus.Pending;

    public string? Observations { get; set; }

    public DateTime? DeletedAt { get; set; }

    public RawMaterialType RawMaterialType { get; set; } = null!;
    public Provider Provider { get; set; } = null!;
    public RawMaterialCharacteristic RawMaterialCharacteristic { get; set; } = null!;

    public MaterialTest? Test { get; set; }
    public ICollection<ProviderCoilMovement> CoilMovements { get; set; } = new List<ProviderCoilMovement>();

    [NotMapped]
    public int ReturnedCoilsCount => CoilMovements?.Sum(c => c.CoilsReturned) ?? 0;
}
